lib/main.dart

```diff
@@ -1,177 +1,537 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 
-import 'menu.dart';
+import 'bloc/todo_bloc.dart';
+import 'breakpoints.dart';
+import 'models/item.dart';
+import 'models/stopwatch.dart';
 
-// Widget codes used primarily for testing
-const appBarKey = Key('appbar');
-const textfieldKey = Key('textfield');
-const saveButtonKey = Key('save_button');
-const iconKey = Key('menu_icon_button');
+// Keys used for testing
+final textfieldKey = UniqueKey();
+final textfieldOnNewPageKey = UniqueKey();
+final saveButtonKey = UniqueKey();
+final itemCardWidgetKey = UniqueKey();
+final itemCardTimerButtonKey = UniqueKey();
+final backButtonKey = UniqueKey();
+final logoKey = UniqueKey();
```

Comment:
what is the purpose of using such keys? are they really needed? unique keys should only be used in cases where maintaining the widget state in the parent widget is really needed (e.g., in a list view that can reorder)

Reply:
for testing you can use a simple `Key` or you can simply rely on `find.byType` API

Reply:
Yeah, these keys are solely used to make testing much easier. 
I've had many run-ins with issues with `find.byType` and the usual `find.text`, where instances were not found and it wasn't discernable during the tests.
I went with this route because it makes implementing tests easier and also reading them. 

If this is blasphemous in the Flutter holy book, please make me aware. Or if there's a better idiomatic way of doing this, I'd be happy to know 😄

Reply:
If you want to go the key path, use simple `Key` instead of `UniqueKey`. Unique key should **only** be used in cases where maintaining a state in the widget tree is something under these lines: 

"I know who I am, I may be removed now from the tree, but when I comeback I will be the same, I don't forget my roots" (pardon the pun)

Reply:
Alrighty!
I'll use `Key('whatever_text')` then.
Thanks!